File: test_opencv.cpp
#include <opencv2/opencv.hpp>


int main() {
    cv::VideoCapture cap(0);  // 创建VideoCapture对象，参数为摄像头索引号，0表示默认摄像头

    if (!cap.isOpened()) {
        std::cout << "无法打开摄像头" << std::endl;
        return -1;
    }

    cv::namedWindow("Camera", cv::WINDOW_NORMAL);  // 创建一个窗口

    while (true) {
        cv::Mat frame;
        cap.read(frame);  // 读取摄像头图像

        if (frame.empty()) {
            std::cout << "无法读取摄像头图像" << std::endl;
            break;
        }

        cv::imshow("Camera", frame);  // 在窗口中显示图像

        // 按下 'q' 键退出循环
        if (cv::waitKey(1) == 'q') {
            break;
        }
    }

    cv::destroyAllWindows();  // 关闭窗口

    return 0;
}

File: test_opencv.py
import cv2

def main():
    # 创建VideoCapture对象，参数为摄像头索引号，0表示默认摄像头
    cap = cv2.VideoCapture(0)

    while True:
        # 读取摄像头图像
        ret, frame = cap.read()

        # 检查图像是否读取成功
        if not ret:
            print("无法读取摄像头图像")
            break

        # 在窗口中显示图像
        cv2.imshow('Camera', frame)

        # 按下 'q' 键退出循环
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break

    # 释放摄像头资源和关闭窗口
    cap.release()
    cv2.destroyAllWindows()

if __name__ == '__main__':
    main()
